h/services/search_index/service.py

```diff
@@ -77,15 +76,7 @@ def add_annotations_between_times(self, start_time, end_time, tag):
         :type start_time: datetime.datetime
         :type end_time: datetime.datetime
         """
-        self._queue.add_all(
-            [
-                row[0]
-                for row in self._db.query(Annotation.id)
-                .filter(Annotation.updated >= start_time)
-                .filter(Annotation.updated <= end_time)
-            ],
-            tag=tag,
-        )
+        self._queue.add_annotations_between_times(start_time, end_time, tag)
```

Comment:
Yep. No need to get fancy. Until we hit the rule of three probably not worth generalising here.